src/aggregator/aggregator/forwarded_writer.go

```diff
@@ -394,16 +434,20 @@ func (agg *forwardedAggregation) reset() {
 // add adds the aggregation key to the set of aggregations. If the aggregation
 // key already exists, its ref count is incremented. Otherwise, a new aggregation
 // bucket is created and added to the set of aggregtaions.
-func (agg *forwardedAggregation) add(key aggregationKey) {
+func (agg *forwardedAggregation) add(metric Registerable) {
+	key, _ := metric.ForwardedAggregationKey()
```

Comment:
nit: Should this be checked too? Guess hard since no error could return from this method...